com.unity.netcode.adapter.utp/Runtime/UTPTransport.cs

```diff
@@ -0,0 +1,835 @@
+        {
+            var eventType = m_Driver.PopEvent(out var networkConnection, out var reader);
+
+            switch (eventType)
+            {
+                case Networking.Transport.NetworkEvent.Type.Connect:
+                    InvokeOnTransportEvent(UTPNetworkEvent.Connect,
+                        ParseClientId(networkConnection),
+                        default(ArraySegment<byte>),
+                        Time.realtimeSinceStartup);
+                    return true;
+
+                case Networking.Transport.NetworkEvent.Type.Disconnect:
+                    InvokeOnTransportEvent(UTPNetworkEvent.Disconnect,
+                        ParseClientId(networkConnection),
+                        default(ArraySegment<byte>),
+                        Time.realtimeSinceStartup);
+                    return true;
+
+                case Networking.Transport.NetworkEvent.Type.Data:
+                    var isBatched = reader.ReadByte();
+                    if (isBatched == 1)
+                    {
+                        while (reader.GetBytesRead() < reader.Length)
+                        {
+                            var payloadSize = reader.ReadInt();
+                            ReadData(payloadSize, ref reader, ref networkConnection);
+                        }
+                    }
+                    else // If is not batched, then read the entire buffer at once
+                    {
+                        var payloadSize = reader.ReadInt();
+
+                        ReadData(payloadSize, ref reader, ref networkConnection);
+                    }
+
+                    return true;
+            }
+
+            return false;
+        }
+
+        private unsafe void ReadData(int size, ref DataStreamReader reader, ref NetworkConnection networkConnection)
+        {
+            if (size > m_MessageBufferSize)
+            {
+                Debug.LogError("The received message does not fit into the message buffer");
+            }
+            else
+            {
+                unsafe
+                {
+                    fixed (byte* buffer = &m_MessageBuffer[0])
+                    {
+                        reader.ReadBytes(buffer, size);
+                    }
+                }
+
+                InvokeOnTransportEvent(UTPNetworkEvent.Data,
+                    ParseClientId(networkConnection),
+                    new ArraySegment<byte>(m_MessageBuffer, 0, size),
+                    Time.realtimeSinceStartup
+                );
+            }
+        }
+
+        private void Update()
+        {
+            if (m_Driver.IsCreated)
+            {
+                m_Driver.ScheduleUpdate().Complete();
+
+                while (AcceptConnection() && m_Driver.IsCreated)
+                {
+                    ;
+                }
+
+                while (ProcessEvent() && m_Driver.IsCreated)
+                {
+                    ;
+                }
+
+                FlushAllSendQueues();
```

Comment:
Shouldn't we do that before the driver update? This way the driver's send job will be able to immediately send it all on the network. Otherwise we wait until the next `Update` to send anything.

Reply:
cc @Cosmin-B as he wrote this code

Reply:
Moved it before the driver update, thanks!